src/XMakeBuildEngine/Evaluation/Expander.cs

```diff
@@ -170,6 +175,12 @@ internal Expander(IPropertyProvider<P> properties, IItemProvider<I> items, IMeta
             _metadata = metadata;
         }
 
+        public Expander(IPropertyProvider<P> properties, IItemProvider<I> items, IItemFactory<I, I> defaultItemFactory) : this(properties, items)
```

Comment:
Why `public` rather than `internal` like the others?

Will we need an overload with the `defaultItemFactory` _and_ `metadata` arguments?